### DISCUSSION TITLE
is the end z switch the zero point of the 4 corner leveling? If so, how to get below that 0.2 layerhight?

### DISCUSSION THREAD

**Discussion by heidepiek** [The-300lb-Gorilla/Ender-2-Pro-v427#5] [Q&A] [answered]
Tonight we where printing a z offset calibration square. We wanted the z offset -0.05. But i think that is not possible with the current
situation. Because you level the bed with the 4 corner bed leveling option. And when the printer does this, the z endstop is triggert.
So when you set your z offset, it can not get below zero. What are we doing wrong? If i lower the endstop, and set the home offsets. 
We can get a perfect first layer. because we can get below that 0.2mm (say 0.15) But if we than use the 4 corner bed leveling option the nozzle digs in the bed.  Hope i make sens and you understand what i mean.

greetz harrie

**Comment by The-300lb-Gorilla** [comment]
You should only use the four corner bed leveling as part of the overall bed leveling. Once you use the 4 point, you should the use the manual mesh bed leveling. It probes the bed with the nozzle at 9 points, using a piece of paper like the 4 point method, but adjusting the height of the nozzle with the rotary encoder knob. It is like have a BL Touch, but you are setting the adjustment and the amount of friction on the piece of paper. The only time you really need the 4 point over the bed screws is before you run a LCD Manually Assisted 9 point bed level after changing a nozzle, or if you take the heated bed from the Y Carriage for some reason

**Reply by heidepiek** [reply]
I understand what your saying. But i  want to lower the nozzle a little bit to squeeze the filament on the first layer. We are doing a first layer test and it is just not perfect yet. The end switch is triggered with homing. than when you use the 4 corner leveling you can level the bed.
This is not always needed, i know. But when i installed the firmware for the first time, the first thing that you do is level the bed. So we did.
The offsets are good. I did probe the 9 points. So Printing time, but i want to lower the nozzle 0.05. I can only get that done when the printer is printing with z babystepping. But everytime you start a new print, it homes again. So you have to use babystepping again to lower the nozzle when printing again. It makes sense because the zero point is the triggerpoint of the endstop. When i lower the endstop the triggerpoint is al little lower than the top of the bed. So you can go down the zero point of the bed. This makes sense in my head. But maybe i am wrong. But my other printer has a bl touch that is used for probing and is the trigger for his z height. This is always in the plus. if you want to lower your nozzle to get more squish that is not a problem. Because it is way above zero, so you can always go lower. Hope i make  sense, 
and you get what i mean. With the ender 2 pro, i can`t get the filament to squish a litle more. So i get a better first layer. If i lower the endstop this is not a problem. But then i can`t use the 4 corner bed leveling anymore. I know i can do it manualy, but i liked the feature. And than i can`t use it anymore. Wel for now i`m going to bed. It is way my bed time. It`s 00.16, and six o clock my alarm goes off. and than to work. Hobbies are great, but sometimes frustrating!  My iq goes so far, and then it stops. 

greetz harrie

**Reply by The-300lb-Gorilla** [reply]
Goto first menu.
select Move, Home, Level.
Select Bed Leveling Menu
Select Z Offset
Use the rotary encoder knob to adjust the z height offset between the nozzle tip and the bed.

**Comment by The-300lb-Gorilla** [comment]
Goto first menu.
select Move, Home, Level.
Select Bed Leveling Menu
Select Z Offset
Use the rotary encoder knob to adjust the z height offset between the
nozzle tip and the bed.

**Reply by heidepiek** [reply]
Hope you don't think me a dumbass. But I can't seem to get the nozzle down. One more time the sequence with how we set up the printer. First, flash the firmware. Followed by m502 and then m500.  Home the printer. Do a 4 corner leveling. 
And then I think it goes wrong. When leveling the bed, the printer uses the end stop as the 0 point. So that is also the z offset of the printer. Because you use a piece of paper to level. That is say 0.1 to 0.2 thick. So the zero point of the printer is above the bed surface. So the nozzle is always hanging when you home
plus minus 0.2 above the bed.  So I can try all kinds of things, but I can't get the nozzle on the bed, for example!!! Why not, because the 0 point is thus 0.2 above the bed. If I then set the end stop 0.2 down. and then print, I can go up and down with the nozzle. But assuming I put the firmware on the printer, and do 4 corner leveling. 9 point leveling. And set the z offset to 0.2. Then why doesn't the printer respond to turning the z offset with the rotary knob? Not up, and not down.  Really, have tried a few hours of everything now. But no success. Like I said, I can't get the nozzle down! Have read this piece (https://printingatoms.com/ender-3-z-offset/) And I think we are using the Physical Endstop way. So if you want the nozzle closer to the bed you have to move the switch down.  Sorry if I get annoying. But don't get it anymore. Maybe I am doing something wrong, but then that is ignorance.  Or there is something wrong in the software. But I don't think so, because your printer probably works fine.

HELP!!!
Harrie

Translated with www.DeepL.com/Translator (free version)

**Reply by heidepiek** [reply]
incompetence on this side. Dumbass that I am.
Am finally figuring it out. You have to turn on the bed leveling, otherwise the z offset rotary won't work!!! But what I forgot every time was to add in cura at machine settings this line after the G28.
Namely (M420 S1 Z2 ;Force printer to use auto bed leveling setting)
Now I can adjust the z offset. And now it is working fine. And i don`t have to adjust the endswitch at a lower point!
Am really spoiled with that professionel firmware from mriscoc.
With that firmware you can bend everything to your will. Is also a whole wiki on his reposetory, then it becomes a bit easier for an amateur. But most importantly, our ender 2 pro now runs like clockwork now!.  I see you replied just when i was typing this!. I wil watch the video.
But i tried to lower the endswitch, and setting the z offset. Won`t do that again! I damaged the magnetic plate a bit. The nozzle digt in to it!
I got it to print, but was not satisfied. I could`t use the 4 corner leveling than, because the zero point was lower than. and the 4 point corner leveling uses the endswitch as zero, so than it caves in the bed. But like i said, this dummie found it out. Thanks for your help! Respect for the programmers. Nice piece of engineering.

Greetzz harrie

**Comment by The-300lb-Gorilla** [comment, marked as answer]
U kunt uw eindstopschakelaar lager instellen dan het horizontale vlak van
het bed. Wanneer de printer de start-GCODE uitvoert, wordt het mondstuk
boven het bed getild
Bekijk deze video, misschien helpt het
--------------------------------------------------------------------------------------------
You can set your end stop switch lower than the horizontal plane of the
bed. When the printer runs the start GCODE it will lift the nozzle
above the bed
Watch this video, it might help

https://www.youtube.com/watch?v=CtbGsJCiZNw <https://youtu.be/CtbGsJCiZNw>

**Reply by heidepiek** [reply]
I can understand english, that`s not a problem. Writing and speaking it is another thing. You explained it perfectly! sometimes I use a translation program because the sentence structure in dutch is sometimes very different in english. So to avoid confusion I then use the translator. Thanks for the video. 
![first layer](https://user-images.githubusercontent.com/104028471/228951209-5c8d0fa0-e38c-49ba-b1ee-bb785e94f2d1.jpeg)

**Comment by The-300lb-Gorilla** [comment]
Does the Dutch explainitation as well as English help? I have a friend who
works in customer support at Creality in China, and we always use
simplified Chinese to write back and forth.
That first layer looks excellent!! 🤓

Cheers!
The 300lb Gorilla

**Reply by heidepiek** [reply]
mr gorilla, first to answer your question. Yes, the translation to dutch is quite correct. Now my English is pretty good too, so I understood what you were saying. That I use a translation program has to do with the sentence structure. The Dutch-English version sometimes wants to twist things around.  The English Dutch  version on the other hand does a better job! I think I understand English better than you understand Dutch. So you probably can't make anything out of the Dutch text. So you can't check if what it says is correct and linguistically correct. How that works out in Chinese, I have no idea.

Now about the firmware and the printer. I learned from my father that if something does not work, or is broken. And you want to fix it, you need to know how something works. If you know how something works, you can often fix it. Watched your video a few times.
But I think your limiter switch is also adjusted so that when you do a 4 corner leveling, you can always put your double folded paper under the nozzle and the bed. The leveling uses the limiter switch as the zero point. Right? I think it went wrong with me because I forgot the m420 in my slicer. So every time you home, the printer goes to the 0 point (limiter switch) And every time you order the printer to print without telling it to use the mesh it just goes to print at its 0 point. Correct? So my problem was that the bed z didn't work. But suggest you mount the limiter switch 4mm below the bed. Then your 4 corner bed leveling won't work correctly either. Right? Because it homes the printer, and when you use the 4 corner leveling again, it goes to the limiter switch every time you press the button. So I think that was also my problem when I started fiddling with the home offsets and had turned the limiter down.
Another thing i have to understand, home offsets! What is that? When I use the m503 it indicates the offsets with the m206. Those default to 0 with your firmware. 
Now with the terminal I send the x and the y to 0 then the nozzle goes nicely to the beginning of the bed.  If I manually set the nozzle to the beginning of the bed, and the nozzle 0.2 above the bed and I press the rotary button to set home offsets on the display and I give the m206 command again, it shows completely different values.
But still if I then give command with the terminal g01 x0 y0. It goes nicely to the starting point of the bed.  And still the values are different from the offset? Difficult matter. I am far from understanding all that. So that's why it went wrong the first few times. Hope you understand me a little bit? Quite a bit of text, and also with questions.  
But to make a long story short, home the printer. 4 corner leveling with paper and then 9 point bed leveling. And most important! M420 in the machine settings of your slicer!!! And then your firmware works great! Oh yeah, i have got the same version as you. But my menu`s are a little bit different than yours. I think its because i use the one with host enabled. I saw that you can edit the mesh to. I cant! It`s not in the menu. But i think also not neccesary, you saw my first layer. i think it`s oke. Well, this it. Now I'll stop. The misses stands around the corner. We have to go shopping. Have a nice weekend. 

Cheers

Harrie

**Comment by The-300lb-Gorilla** [comment]
Okay.🤓
